### ISSUE TITLE
[QUEST] The Drakkensryd

### ISSUE THREAD

**Issue by click** [TrinityCore/TrinityCore#1350]
The script summons Proto-Drake mount and its way-points is missing. 
It's impossible to fight on the mount when you use Harpoon.
Video: http://www.youtube.com/watch?v=0M2y0bJRBn4

Core Rev: 11600
DB rev: 2011_02_24

**Comment by Nexus-WoW** [comment]
Anyone got some News about "The Drakkensyrd" Quest?

**Comment by DDuarte** [comment]
By @octon:

> Core rev. 48fa7fa

> Quest http://www.wowhead.com/quest=12886

> How it should look like:
https://www.youtube.com/watch?v=0M2y0bJRBn4

> Bugs:
Proto-drake (which flies to the Temple) doesnt work.
Hyldsmeet Drakerider (http://www.wowhead.com/npc=29694) always "out of range".
Column Ornament (http://www.wowhead.com/npc=29754) falls on the floor.

> Cosmetic: 
Thorim (http://www.wowhead.com/npc=29445) sits a little incorrectly.

**Comment by cobralyn** [comment]
1 drake that flies you there need to just be given waypoints to fix this. 2as for the out of range issue it is mostly solved by facing the direction of the flying drake before you  chain onto it. or  just make the dragons fly a little closer to the temple. or just spawn a bunch of riders along the brim of the temple3 29445 just needs the  number 4 in the inhabit type column. 

     On Saturday, January 24, 2015 12:20 PM, Duarte Duarte <notifications@github.com> wrote:
   

 By @octon:
Core rev. 48fa7faQuest http://www.wowhead.com/quest=12886How it should look like:
https://www.youtube.com/watch?v=0M2y0bJRBn4Bugs:
Proto-drake (which flies to the Temple) doesnt work.
Hyldsmeet Drakerider (http://www.wowhead.com/npc=29694) always "out of range".
Column Ornament (http://www.wowhead.com/npc=29754) falls on the floor.Cosmetic: 
Thorim (http://www.wowhead.com/npc=29445) sits a little incorrectly.

**Comment by Rushor** [comment]
probably someone can properly fix it, an old attempt is http://pastebin.com/i21xunk6

**Comment by ccrs** [comment]
- player needs to enter in a drake when mission is taken.
- this mount takes off on a wp path to the place where the mission takes place and returns to the same spot.
- player remains with the disguise all the time.
- when near another drake, player should be able to mount it leaving the initial one.

**Comment by Killyana** [comment]
This  will fix the credit quest, on rev. 68c4ea314443 just the fly from the quest giver to the quest area is missing, every thing else works or me.
``` sql
UPDATE `creature_template` SET `InhabitType`=4 WHERE `entry`=29754;
DELETE FROM `smart_scripts` WHERE `entryorguid`=29694 AND `id`=1;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(29694,0,1,0,6,0,0,0,0,0,0,0,33,29800 ,0,0,0,0,0,7,0,0,0,0,0,0,0,'Hyldsmeet Drakerider - On Death - Kill credit Spell');

**Comment by ccrs** [comment]
The "not in range" issue is still present? Was it solved by changing vehicleid?
Last, the player has the disguise spell active while doing it?

**Comment by Killyana** [comment]
The "not in range" issue still happen sometimes, the disguise spell is fixed long time ago.

**Comment by Nexus-WoW** [comment]
Not in Range only for Meele DDs, does someone know where i can find it?

**Comment by danlapps** [comment]
This would need some more testing for close and pull right ?

**Comment by Killyana** [comment]
Here the script: https://github.com/TrinityCore/TrinityCore/blob/ab90f744866d761a302380e12ed7b0dee23697f0/src/server/scripts/Northrend/zone_storm_peaks.cpp#L370
The problem comes probably from the spawn behavior.

**Comment by w5860363** [comment]
quest_template  target is 29800 
creature spawn is 29694 
The task can not be completed 
Please excuse my English.

**Comment by Runico** [comment]
any news!

**Comment by ccrs** [comment]
its not from script, i think it is a general vehicle issue.

**Comment by DearAdmin** [comment]
after this : https://github.com/TrinityCore/TrinityCore/issues/1350#issuecomment-78263791 

no mount up but drake is there. ,

![bug](https://cloud.githubusercontent.com/assets/10039561/10016956/6cc6df26-6133-11e5-921a-066d1ebf88db.png)

**Comment by Killyana** [comment]
It works for me, your core is probably outdated as I removed the hack used for the vehicle.

**Comment by DearAdmin** [comment]
yeah @Killyana  rev is TrinityCore rev. f50df72

**Comment by DevQuad** [comment]
The problem with this quest is the movement of the dragon. If we set 0 MovementType the entire script works fine.

**Comment by tkrokli** [comment]
@DevQuad : that sounds interesting, are you referring to the Hyldsmeet Drakerider mount, since those vehicles stay in a constant "loop" around the tower?

**Comment by DevQuad** [comment]
Yes, I mean this mount, if you stop them to move on at some point use a harpoon mobs are included in the combat and fighting.

**Comment by tkrokli** [comment]
Oh. Right. Then that will not be a proper solution, because the original behaviour is that **both rider and mount should continue to move around the tower during the fight** as well as outside of combat.

**Comment by DevQuad** [comment]
I say the problem is the inability to attack the rider is in the movement of the dragon. That is, the problem must be sought in the motion.

**Comment by tkrokli** [comment]
That would make sense, I think. It also makes me think about the visual issue in Icecrown where the NPCs on Orgrim's Hammer drag behind the ship when your own character is flying near the ship (before landing). I don't know enough about how the units are controlled (like in Unit.cpp), how they are linked to their vehicles and such, so we need to find out where this problem starts. A year ago, Killyana commented that this problem probably comes from the spawn behavior in https://github.com/TrinityCore/TrinityCore/blob/2da458c/src/server/scripts/Northrend/zone_storm_peaks.cpp#L371 -- read the script and try to find out what could be causing this problem, see if Killyana's theory about spawning is correct.

**Comment by sirikfoll** [comment]
I don't think is the spawn, if you stop the drake, install the accessory and then let the drake move, still don't work.
What matters here is if the drake is moving.
If you do this:
- 1 Stop a drake and mount up. Everything's fine until the drake moves, then it bugs again("you are too far" error)
Some position is not getting updated correctly, no idea where and why..

**Comment by tkrokli** [comment]
Could it be something wrong with the passenger seat of the vehicle, that the hostile NPC is visually there, but the core does not update the passenger position because there is a seat conflict?

**Comment by sirikfoll** [comment]
just some feedback, this is exactaly the same issue as the old one "player on Lord Marrowgar's Bone spike can't be healed", something with the calculus of the passenger offset position.

Sniffing the correct positions for both seats and using the same workaround used for the bone spike([this](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/IcecrownCitadel/boss_lord_marrowgar.cpp#L469-L475)) fix this issue, but is a hack anyway.

**Comment by Txknight** [comment]
I made this quest work, it is not perfect, swapping the seat of the drake rider from 1 to 0 places the rider on the back of the dragon, and you in the saddle allowing you to turn around and kill, no more range issues. I put in way points, changed Gretta to summon a proper hyldsmeet drake, not the frigid one from another attempted fix, the drake flies you up there, circles twice and returns to Gretta. Possibly my vehicle accessory and vehicle template are duplicated, may not need both. Changed the rider from casting mortal strike to a unarmed melee attack, since they have no weapons, hard to cast mortal strike. Hope this helps, if I have read the vehicleseat.dbc correct, the two seats are supposed to be 3.5 apart, but when I would use a macro npc info and .gps myself and the target occupied the same exact coordinates. but swapping the seat location fixed it, wonky fix, but the quest works, enjoy!

[Drakkensryd SQL.txt](https://github.com/TrinityCore/TrinityCore/files/230065/Drakkensryd.SQL.txt)
